### ISSUE TITLE
Custom toolchain; unable to find libmpv

### ISSUE THREAD

**Issue by DeadSix27** [u8sand/Baka-MPlayer#231]
Trying to compile this in my custom toolchain, libmpv is compiled like:

`./waf configure --enable-sdl2 --enable-libmpv-static`

However when I configure Baka-MPlayer
I get this error:

"Project ERROR: mpv development package not found"

Config line: (Note I setup a MXE toolchain just to compile QT5 and use a different toolchain for mpv and try to use it for Baka-Mplayer)
```

		QTROOT="/mxe/mxe/usr/x86_64-w64-mingw32.static/qt5/bin"
		QMAKE="$QTROOT/qmake" \
		./configure \
		"CONFIG+=embed_translations" \
		"lupdate=$QTROOT/lupdate" \
		"lrelease=$QTROOT/lrelease"
```
pkg-config finds it, shouldnt qmake too? Thats the part that confuses me:
```
$ pkg-config --print-provides mpv
mpv = 1.23.0
$ pkg-config --libs mpv
-L/f/toolchain/mingw-w64-x86_64/lib -lmpv
```

Any idea?

**Comment by u8sand** [comment]
In the past, the primary issue with pkgbuild was mpv's pkgbuild not being
generated properly.
https://github.com/u8sand/Baka-MPlayer/blob/master/windows/cross-compilation/baka-build.sh#L113-L117

The fact that your pkg-config is working suggests otherwise. Perhaps it has
something to do with an internal path of mxe's qmake. It may be using a
different pkg-config than the one you intend. Ensure you're properly hiding
everything in your path for the toolchain:
https://github.com/u8sand/Baka-MPlayer/blob/master/windows/cross-compilation/baka-build.sh#L88-L98

Hopefully that helps

**Comment by DeadSix27** [comment]
@u8sand My paths are what seems to be correct in this situation I think:

```
PKG_CONFIG_PATH = /f/toolchain/mingw-w64-x86_64/x86_64-w64-mingw32/lib/pkgconfig
PATH = /f/toolchain/mingw-w64-x86_64/bin:/home/****/bin:/home/****/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/opt/extra/bin:/snap/bin
```

I will look into qmake and whether it has a special env var. for PKG CONFIG (Altho I think I searched that before and couldn't find anything)

**Comment by u8sand** [comment]
Note that I've recently begun using this forked mxe repo: https://github.com/u8sand/mxe
I've added mpv and baka-mplayer Makefiles to allow mxe to build baka by itself: (make baka-mplayer)

```
ARCH=(i686 x86_64)

if [ ! -d release ]; then
  mkdir -p release{,/{mpv,fonts}}

  cd release
  curl -L https://yt-dl.org/downloads/latest/youtube-dl.exe -o youtube-dl.exe

  cd fonts
  curl https://noto-website-2.storage.googleapis.com/pkgs/NotoSansSymbols-unhinted.zip | bsdtar -xvf -
  curl https://noto-website-2.storage.googleapis.com/pkgs/NotoSans-hinted.zip | bsdtar -xvf -
  cd ..

  cd ..
fi

if [ ! -d build ]; then
  git clone git@github.com:u8sand/mxe.git build
  cat << EOF > build/settings.mk
JOBS := $(grep -c ^processor /proc/cpuinfo)
MXE_TARGETS := ${ARCH[@]/%/-w64-mingw32.static}
EOF
  cd build && make baka-mplayer
else
  cd build && git pull
  cd build && make update
fi

function upx_or_cp {
  upx $1 -o $2 || cp $1 $2
}

for arch in $ARCH; do
  mkdir -p $arch
  cp -r release/* $arch
  upx_or_cp "build/usr/$arch-w64-mingw32.static/bin/baka-mplayer.exe" "$arch/baka-mplayer.exe"
  cd $arch
  zip "../$arch.zip" -r *
  cd ..
  rm -r $arch
done
```

**Comment by DeadSix27** [comment]
The main deal being that I do not want to use MXE for MPV, FFMPEG etc, I use a custom toolchain script i forked of ffmpeg helpers, which is highly customized.

What  I tried to do is compile JUST qt5 with mxe ('cuz thats a pain to do..) everything else I compile within my own toolchain. 

Now with the last edit of my previous post I figured out how to compile your UI in my toolchain and it works just fine, Today I reset my entire toolchain and will check whether everything will work out automatically without issues.
